spring-cloud-gcp-secretmanager/src/main/java/org/springframework/cloud/gcp/secretmanager/SecretManagerPropertySource.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2017-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.secretmanager;
+
+import com.google.cloud.secretmanager.v1beta1.AccessSecretVersionResponse;
+import com.google.cloud.secretmanager.v1beta1.SecretManagerServiceClient;
+import com.google.cloud.secretmanager.v1beta1.SecretVersionName;
+import com.google.protobuf.ByteString;
+
+import org.springframework.cloud.gcp.core.GcpProjectIdProvider;
+import org.springframework.core.env.EnumerablePropertySource;
+
+/**
+ * A property source for Secret Manager which accesses the Secret Manager APIs when {@link #getProperty} is called.
+ *
+ * @author Daniel Zou
+ * @author Eddú Meléndez
+ * @since 1.2.2
+ */
+public class SecretManagerPropertySource extends EnumerablePropertySource<SecretManagerServiceClient> {
+
+	private final GcpProjectIdProvider projectIdProvider;
+
+	public SecretManagerPropertySource(
+			String propertySourceName,
+			SecretManagerServiceClient client,
+			GcpProjectIdProvider projectIdProvider) {
+		super(propertySourceName, client);
+
+		this.projectIdProvider = projectIdProvider;
+	}
+
+	@Override
+	public Object getProperty(String name) {
+		SecretVersionName secretIdentifier =
```

Comment:
Can we wire in the template and use it here?

Reply:
Done.